docs/concept/buckconfig.soy

```diff
@@ -2006,6 +2007,87 @@ cxx_library(
   {/param}
 {/call}
 
+{call buckconfig.section}
+  {param name: 'kotlin' /}
+  {param description}
+    This section configures various aspects of the Kotlin toolchain.
+  {/param}
+{/call}
+
+{call buckconfig.entry}
+  {param section: 'kotlin' /}
+  {param name: 'kotlinc' /}
+  {param example_value: '/usr/local/bin/kotlinc' /}
+  {param description}
+    The path to the <code>kotlinc</code> compiler executable to use when external compilation is
+    forced.  This setting has no effect by itself and must be paired with the <code>external</code>
+    {sp}setting.
+  {/param}
+{/call}
+
+{call buckconfig.entry}
+  {param section: 'kotlin' /}
+  {param name: 'external' /}
+  {param example_value: 'true' /}
+  {param description}
+    Forces external compilation via <code>kotlinc</code>.  When external compilation is forced the
+    following heuristics are used to locate the <code>kotlinc</code> executable:
+    <ul>
+      <li>If the <code>kotlinc</code> setting is specified, the executable specified by that path
+          will be used.
+      </li>
+      <li>If the <code>kotlin_home</code> path setting is specified, Buck will look for a
+          {sp}<code>bin</code> directory under that path for an executable named
+          {sp}<code>kotlinc</code>.
+      </li>
+      <li>If a <code>KOTLIN_HOME</code> environment variable is present, Buck will look for a
+          {sp}<code>bin</code> directory under that path for an executable named
+          {sp}<code>kotlinc</code>.
+      </li>
+      <li>Lastly, if none of the above are specified, Buck will look for the <code>kotlinc</code>
+          {sp}executable in the paths listed in the <code>PATH</code> environment variable.
+      </li>
+    </ul>
+    Defaults to <code>false</code>.
+  {/param}
+{/call}
+
+{call buckconfig.entry}
+  {param section: 'kotlin' /}
+  {param name: 'kotlin_home' /}
+  {param example_value: '/usr/local/Cellar/kotlin/1.1.1' /}
+  {param description}
+    The path to the Kotlin root folder, typically the installation folder, where various Kotlin
+    assets (executables and JAR files) can be found.  This path is used in the following ways:
+    <ul>
+      <li>
+        When in-memory compilation is used, the <code>kotlin-compiler.jar</code> and
+        {sp}<code>kotlin-stdlib.jar</code> are located via this path using the following heuristics:
```

Comment:
Do we want to mention about the fallback for kotlin-runtime for older versions? If not, it might make sense to remove that fallback in a followup change